test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs

```diff
@@ -57,6 +57,21 @@ public void ParseRouteParameter_ConstraintAndOptional_ParsedCorrectly()
         }
 
         [Fact]
+        public void ParseRouteParameter_ConstraintAndOptional_WithDefaultValue_ParsedCorrectly()
+        {
+            // Arrange & Act
+            var templatePart = ParseParameter(@"param:int=12?");
```

Comment:
What happens if we double up the question mark? What about one of the other delimiters.